components/server/src/workspace/workspace-starter.ts

```diff
@@ -593,6 +597,10 @@ export class WorkspaceStarter {
                         return;
                     }
                 }
```

Comment:
Looks like moving these lines up before the image build code, would do as well? The code seems a bit complicated and grown with the conditional execution of a function in two places. Are you concerned that changing the order would break something?

Reply:
Nice find btw. The discussed token optimization wouldn't have helped with this code in place 🙄

Reply:
> The discussed token optimization wouldn't have helped with this code in place

:100: And it's better behavior in general, so nice to have it in anyway.